docs/core/extensions/snippets/channels/Program.Consumer.cs

```diff
@@ -0,0 +1,39 @@
+﻿internal static partial class Program
+{
+    // <awaitforeach>
+    static async ValueTask ConsumeWithAwaitForeachAsync(
+        ChannelReader<Coordinates> reader)
+    {
+        await foreach (Coordinates coordinates in reader.ReadAllAsync())
+        {
+            Console.WriteLine(coordinates);
+        }
+    }
+    // </awaitforeach>
+
+    // <whiletrue>
+    static async ValueTask ConsumeWithWhileAsync(
+        ChannelReader<Coordinates> reader)
+    {
+        while (true) // Only valid with never ending (infinite) producer.
```

Comment:
It's valid even without that, but then the ReadAsync call could end up throwing a ChannelClosedException which would need to be dealt with.

Reply:
Updating comment